### DISCUSSION TITLE
Next.js <Link /> navigation causes session logout on Django page

### DISCUSSION THREAD

**Discussion by mrxsal** [vercel/next.js#53628] [Help] [answered]
### Summary

I am working on an application that combines both Next.js and Django (using django-nextjs package). The authentication is managed by Django using session-based authentication. Here's the sequence of events leading to the issue:

  1.  From the homepage (a Next.js page) where I'm not logged in, I navigate to a Django-based login page.
  2. After successfully logging in, I get redirected to a Next.js page. So far, everything works correctly, and I can access user data from the session cookie.
  3. As long as I navigate between Next.js pages, the session remains active, and I remain logged in.
  4. If I manually enter a URL to a Django page that requires authentication or refresh the page, it recognizes my session, and I remain logged in.
  5. However, if I navigate to a Django page using the Next.js <Link /> component, I get logged out instantly.

What's puzzling is that the session seems active and recognized across both frameworks until I use the Next.js component to navigate to a Django page. When I do this, Django no longer recognizes my session.

What could be causing the session to be lost when navigating via the component? Has anyone faced a similar issue, and how can I fix this?

### Additional information

I am using Next JS 13 app router.
This only happens in the production environment. In development, the session remains active, and there are no issues with authentication when navigating via the <Link /> component.

**Comment by danialkeimasi** [comment, marked as answer]
Hi, I am maintainer of [django-nextjs](https://github.com/QueraTeam/django-nextjs).
You should probably consider using regular `<a />` tag to navigate to Django pages, since client-side routing behavior of `<Link />` only works when the target page is Nextjs page.

**Reply by mrxsal** [reply]
Thanks, this is indeed what I ended up doing.
